src/resonanceReconstruction/rmatrix/legacy/unresolved/CompoundSystem/test/evaluate.test.hpp

```diff
@@ -0,0 +1,1114 @@
+    ResonanceTable table22(
+      { Resonance( 1.5e+4 * electronVolt, 55793.70 * electronVolt, 36.82380 * rootElectronVolt, 1.081650 * electronVolt, 0. * electronVolt, 0. * electronVolt ),
+        Resonance( 1.0e+5 * electronVolt, 55793.70 * electronVolt, 36.82380 * rootElectronVolt, 1.081650 * electronVolt, 0. * electronVolt, 0. * electronVolt ) },
+      { 2, 0, 0, 0 } );
+    ResonanceTable table23(
+      { Resonance( 1.5e+4 * electronVolt, 68407.20 * electronVolt, 45.14870 * rootElectronVolt, 1.081650 * electronVolt, 0. * electronVolt, 0. * electronVolt ),
+        Resonance( 1.0e+5 * electronVolt, 68407.20 * electronVolt, 45.14870 * rootElectronVolt, 1.081650 * electronVolt, 0. * electronVolt, 0. * electronVolt ) },
+      { 2, 0, 0, 0 } );
+    ResonanceTable table24(
+      { Resonance( 1.5e+4 * electronVolt, 102952.0 * electronVolt, 67.94820 * rootElectronVolt, 1.081650 * electronVolt, 0. * electronVolt, 0. * electronVolt ),
+        Resonance( 1.0e+5 * electronVolt, 102952.0 * electronVolt, 67.94820 * rootElectronVolt, 1.081650 * electronVolt, 0. * electronVolt, 0. * electronVolt ) },
+      { 2, 0, 0, 0 } );
+    ResonanceTable table25(
+      { Resonance( 1.5e+4 * electronVolt, 185730.0 * electronVolt, 61.29090 * rootElectronVolt, 1.081650 * electronVolt, 0. * electronVolt, 0. * electronVolt ),
+        Resonance( 1.0e+5 * electronVolt, 185730.0 * electronVolt, 61.29090 * rootElectronVolt, 1.081650 * electronVolt, 0. * electronVolt, 0. * electronVolt ) },
+      { 1, 0, 0, 0 } );
+
+    // spin groups
+    SpinGroup group00( std::move( elastic00 ), std::move( table00 ) );
+    SpinGroup group01( std::move( elastic01 ), std::move( table01 ) );
+    SpinGroup group10( std::move( elastic10 ), std::move( table10 ) );
+    SpinGroup group11( std::move( elastic11 ), std::move( table11 ) );
+    SpinGroup group12( std::move( elastic12 ), std::move( table12 ) );
+    SpinGroup group13( std::move( elastic13 ), std::move( table13 ) );
+    SpinGroup group20( std::move( elastic20 ), std::move( table20 ) );
+    SpinGroup group21( std::move( elastic21 ), std::move( table21 ) );
+    SpinGroup group22( std::move( elastic22 ), std::move( table22 ) );
+    SpinGroup group23( std::move( elastic23 ), std::move( table23 ) );
+    SpinGroup group24( std::move( elastic24 ), std::move( table24 ) );
+    SpinGroup group25( std::move( elastic25 ), std::move( table25 ) );
+
+    // the compound system
+    CompoundSystem system( { group00, group01, group10, group11, group12,
+                             group13, group20, group21, group22, group23,
+                             group24, group25 } );
+
+    ReactionID elas( "n,Na22->n,Na22" );
+    ReactionID capt( "n,Na22->capture" );
+
+    THEN( "cross sections can be calculated" ) {
+
+      std::map< ReactionID, CrossSection > xs;
+      system.evaluate( 1.5e+4 * electronVolt, xs );
+      CHECK( 2 == xs.size() );
+      CHECK( 9.3027405025178052 == Approx( xs[ elas ].value ) );
```

Comment:
Where did you get these test values? Also, that's far more precision than Catch can handle with a simple `Approx(...)`.

Reply:
NJOY2016 intermediate output values, that's what it says at the top of the hpp file.